File: consumer.js
const amqp = require('amqplib');

async function connect() {
    const connection = await amqp.connect('amqp://user:password@localhost:5672');
    const channel = await connection.createChannel();
    return { connection, channel };
}

async function receiveMessage(queue, callback) {
    const { channel } = await connect();
    await channel.assertQueue(queue);
    channel.consume(queue, (message) => {
        if (message !== null) {
            callback(message.content.toString());
            channel.ack(message);
        }
    });
}

// Ejemplo de uso:
const queue = 'mi_cola';

receiveMessage(queue, (message) => {
    console.log(`Mensaje recibido: ${message}`);
});

File: consumer.py
import pika

def callback(ch, method, properties, body):
    print(f" [x] Received {body}")

credentials = pika.PlainCredentials('user', 'password')
connection = pika.BlockingConnection(pika.ConnectionParameters(
    'localhost',
    5672,  
    '/',
    credentials))

channel = connection.channel()

channel.queue_declare(queue='hello')

def callback(ch, method, properties, body): 
    print(f" [x]  Received {body}")

channel.basic_consume(queue='hello', on_message_callback=callback, auto_ack=True)

print(' [*] Waiting for messages. To exit press CTRL+C')
channel.start_consuming()
